### PR TITLE
Add application VERSION file

### PR DESCRIPTION
**Pull request by cdwilson**
I wasn't sure if this is something that had been considered yet and wanted to see if it might be worth adopting for the RD template (or if there are reasons why this is a bad idea for now).

The [nRF Connect SDK v2.5.0 Release Notes](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/releases_and_maturity/releases/release-notes-2.5.0.html#build-system) recommend using a dedicated `VERSION` file to set the app/MCUboot version (vs. setting it with the `CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION` Kconfig option)

This PR adds an app `VERSION` file and updates the boot-up logging output to display the following:

- App version (`<VERSION_MAJOR>.<VERSION_MINOR>.<PATCHLEVEL>-<EXTRAVERSION>`, e.g. `1.2.3-rc0`)
- App build version (git rev, i.e. the output of `git describe --abbrev=12 --always`, e.g. `0bd13ef3dff6`)
- MCUboot version used for signing (`<VERSION_MAJOR>.<VERSION_MINOR>.<PATCHLEVEL>`, e.g. `1.2.3`)
- Modem firmware version (read out from the nRF9160 SIP)

Here's some example output from firmware built using the command `west build -p -b nrf9160dk_nrf9160_ns app`:

```
[00:00:00.512,664] <inf> golioth_rd_template: App version: 0.0.0
[00:00:00.512,695] <inf> golioth_rd_template: App build version: 0bd13ef3dff6
[00:00:00.512,725] <inf> golioth_rd_template: MCUboot version: 0.0.0+0
[00:00:00.541,351] <inf> golioth_rd_template: Modem firmware version: mfw_nrf9160_1.3.5
```

One thing worth considering is that specifying version info via the CLI (e.g. `  -- -DCONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION=\"99.99.99\"`) only overrides the MCUboot version used for signing, but does not override the app version info:

```
[00:00:00.512,634] <inf> golioth_rd_template: App version: 0.0.0
[00:00:00.512,664] <inf> golioth_rd_template: App build version: 0bd13ef3dff6
[00:00:00.512,695] <inf> golioth_rd_template: MCUboot version: 99.99.99
[00:00:00.541,320] <inf> golioth_rd_template: Modem firmware version: mfw_nrf9160_1.3.5
```

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -30,6 +31,11 @@ LOG_MODULE_REGISTER(golioth_rd_template, LOG_LEVEL_DBG);
 #include <modem/modem_info.h>
 #endif
 
+/* Macrobatics to convert APP_BUILD_VERSION to a string */
+#define _STR(s)			 #s
+#define STR(s)			 _STR(s)
+#define APP_BUILD_VERSION_STRING STR(APP_BUILD_VERSION)
+
```
Comment (cdwilson) [inline src/main.c:38]: I'm not sure why `APP_VERSION_STRING` is quoted in `app_version.h` but `APP_BUILD_VERSION` is not... (see https://docs.zephyrproject.org/latest/build/version/index.html#use-in-code for details)

It feels like there should be a better way to print the `APP_BUILD_VERSION` as a string, but I'm not aware of one.

**Comment by szczys** [comment]
I think we should do this in the prj.conf file instead of adding a dedicated file for it.

**Comment by cdwilson** [comment]
Is there a way to specify this in the `prj.conf`? I was looking at https://docs.zephyrproject.org/latest/build/version/index.html and didn't see that mentioned in there.

**Comment by szczys** [comment]
You should be able to simply use that symbol in prj.conf instead of passing it on the build command. I believe the build command will still be able to override the prj setting so that's a bonus too (CI, etc)

**Comment by cdwilson** [comment]
Ahh, I see what you mean, yes, we can just specify `CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION` in the `prj.conf` for the app.

> I believe the build command will still be able to override the prj setting so that's a bonus too (CI, etc)

Yes, that's correct. However, if we were to store the version in the `VERSION` file, is there a reason why we would want to be able to override it at build time in CI? I think this change would remove the need for CI to specify the version in the first place since it would already be stored in the release commit that's checked out during the build.

For comparison, there's a couple additional benefits I think we would get with the `VERSION` file approach:

1. If we only use `CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION` (vs. using the `VERSION` file), we won't have access to the Git SHA (`APP_BUILD_VERSION`) used to build the image. This is useful for traceability/debugging (even during development) to make sure somebody didn't accidentally build an image from the wrong Git commit.
2. The `VERSION` file is checked into Git so I don't think we would need CI to specify the version as a command line argument when doing a release. In the case of somebody doing a release manually, it would reduce the chances of them making a typo in specifying the version.
3. The v2.5.0 release notes mentioned "using a VERSION file is the recommended approach" (vs. setting `CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION`)

That said, I don't think there's anything wrong with using `CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION` so I'm good with either approach! (just wanted to show another possible option)

**Pull request closed without merging**